src/Data/BoolExpr.hs

```diff
@@ -0,0 +1,386 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -fno-warn-missing-methods #-}
+
+--------------------------------------------------------------------
+
+--------------------------------------------------------------------
+
+-- |
+-- Module    : Data.BoolExpr
+-- Copyright : (c) Nicolas Pouillard 2008,2009
+-- License   : BSD3
+--
+-- Maintainer: Nicolas Pouillard <nicolas.pouillard@gmail.com>
```

Comment:
I think it would be better to fork and depend on the commit.

@kostmo did you make a PR to the original package?

If it's stale we can figure out a solution before next Swarm release.

Reply:
> I think it would be better to fork and depend on the commit.

Not sure what this means, can you elaborate?

> @kostmo did you make a PR to the original package?

Yes, [here it is](https://github.com/np/boolexpr/pull/2).  I also emailed the maintainer about making another Hackage release on Dec. 27, but haven't heard back yet.

Reply:
Note that package was last uploaded to Hackage in 2011, and the last git commit is from 5 years ago, not very promising.  However I think Nicolas is still around.

Reply:
@kostmo we used to do this for `hsnoise`:

https://github.com/swarm-game/swarm/blob/03509d7ae8366f9a07de819816df8ce0099e0d1e/cabal.project#L4-L7

We got rid of it in #746 as we discovered it was causing issues in the released version:
- #745

Reply:
It seems that using `source-repository-package` carries some complexity cost and risk for Hackage releases. "phadej" [comments](https://github.com/haskell/cabal/issues/8039#issuecomment-1064433772) with a recommendation to "vendor" dependencies / "keep it simple".

In this PR, I've stripped the dependency on `boolexpr` to about half of a single module and copied this into the `swarm` source tree.  I feel like that's a good solution until the long-term maintainership of the `boolexpr` package on Hackage is resolved.

Reply:
Seems like a good solution to me as well.